protocol/encryption/protocol.go

```diff
@@ -422,6 +423,48 @@ func (p *Protocol) BuildHashRatchetMessage(groupID []byte, payload []byte) (*Pro
 	return spec, nil
 }
 
+func (p *Protocol) EncryptRecipientGrants(privateKey *ecdsa.PrivateKey, recipientGrants map[*ecdsa.PublicKey][]byte) (map[uint32][]byte, error) {
+	grants := make(map[uint32][]byte)
+
+	for recipientKey, grant := range recipientGrants {
+		sharedKey, err := generateSharedKey(privateKey, recipientKey)
+		if err != nil {
+			return nil, err
+		}
+
+		encryptedGrant, err := encrypt(grant, sharedKey, rand.Reader)
+		if err != nil {
+			return nil, err
+		}
+
+		kBytes := publicKeyMostRelevantBytes(recipientKey)
+
+		if grants[kBytes] == nil {
+			grants[kBytes] = encryptedGrant
+		} else {
+			grants[kBytes] = append(grants[kBytes], encryptedGrant...)
+		}
```

Comment:
```suggestion
		grants[kBytes] = encryptedGrant
```

I don't think it makes sense in that case. There is one grant per member.

Reply:
fixed